api/authentication.go

```diff
@@ -84,21 +85,42 @@ func userLogin(c *fiber.Ctx) error {
 
 	mapUserLoginPayloadToDbUser(&user, &dbUser) //maping user to db obj
 
-	if err := db.GetUserByEmailAndPassword(&dbUser); err != nil { // sending it to db
+	if err := app.UserLogin(&dbUser); err != nil { // sending it to db
 
-		logger.Error("login err", err)
+		logger.Error("login err <?>", err)
 
 		return c.Status(404).JSON(err.Error())
 	}
 
-	return c.Status(200).JSON(dbUser.ID)
+	token, err := app.CreateJWT(dbUser.ID)
+	if err != nil {
+		return c.Status(400).JSON(err.Error())
+	}
+	return c.Status(200).JSON(token)
+
+}
+func me(c *fiber.Ctx) error {
 
+	tokenString := c.Params("token")
```

Comment:
get token from `Headers` use something like `X-Authorization` header

Reply:
```func me(c *fiber.Ctx) error {

	var tokenString = c.Get("X-Auth-Token", "null")
	id, err := app.ParseJWT(tokenString)
	if err != nil {
		logger.Error("JWT Token Error:<?>", err)
		return c.Status(400).JSON(err.Error())
	}

	user, err := app.GetUser(id)

	if err != nil {
		logger.Error("Get User Error:<?>", err)
		return c.Status(400).JSON(err.Error())
	}
	return c.Status(200).JSON(&user)
}```

Reply:
and why it is needed